pkg/controllers/scheduling/scheduling_controller.go

```diff
@@ -333,6 +333,22 @@ func (c *schedulingController) getEligibleClusterSets(placement *clusterapiv1bet
 		clusterSetNames = clusterSetNames.Intersection(sets.NewString(placement.Spec.ClusterSets...))
 	}
 
+	// ignore the non-legacy clusterset
+	nonLegacyClusterSetNames := sets.NewString()
+	for clusterSetName := range clusterSetNames {
+		clusterSet, err := c.clusterSetLister.Get(clusterSetName)
+		if err != nil {
+			nonLegacyClusterSetNames.Insert(clusterSetName)
+			continue
+		}
+
+		selectorType := clusterSet.Spec.ClusterSelector.SelectorType
+		if len(selectorType) > 0 && selectorType != clusterapiv1beta1.LegacyClusterSetLabel {
+			nonLegacyClusterSetNames.Insert(clusterSetName)
+		}
+	}
+	clusterSetNames = clusterSetNames.Difference(nonLegacyClusterSetNames)
```

Comment:
I do not think we should ignore clusterset here. I'd rather we have a logic in getAvailableClusters that only select clusters from legacy cluster set

Reply:
moved the logic to getAvailableClusters